generic/lazylru.go

```diff
@@ -397,6 +397,26 @@ func (lru *LazyLRU[K, V]) MSetTTL(keys []K, values []V, ttl time.Duration) error
 	return nil
 }
 
+// Delete elimitates a key from the cache. Removing a key that is not in the index is safe.
+func (lru *LazyLRU[K, V]) Delete(key K) {
+	// if the key isn't here, don't bother taking the exclusive lock
+	lru.lock.RLock()
+	_, ok := lru.index[key]
+	lru.lock.RUnlock()
+	if !ok {
+		return
+	}
+	lru.lock.Lock()
+	pqi, ok := lru.index[key]
+	if !ok {
+		return
```

Comment:
This branch needs to unlock the mutex

Reply:
oops! thanks!

Reply:
that would have taken forever to find -- lock up the whole process but only if there is a race condition that you lose when deleting an item multiple times. oof!